mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown

```diff
@@ -876,6 +876,21 @@ linux_node_config {
 
 * `node_group` - (Optional) Setting this field will assign instances of this pool to run on the specified node group. This is useful for running workloads on [sole tenant nodes](https://cloud.google.com/compute/docs/nodes/sole-tenant-nodes).
 
+* `advanced_machine_features` - (Optional) Specifies options for controlling
+  advanced machine features.
+
+<a name="nested_advanced_machine_features"></a>The `advanced_machine_features` block supports:
+
+* `threads_per_core` - (Optional) The number of threads per physical core. To disable simultaneous multithreading (SMT) set this to 1. If unset, the maximum number of threads supported per core by the underlying processor is assumed.
```

Comment:
```suggestion
* `threads_per_core` - (Required) The number of threads per physical core. To disable simultaneous multithreading (SMT) set this to 1. If unset, the maximum number of threads supported per core by the underlying processor is assumed.
```

Reply:
Actually this is optional in the API, so I changed the code to be optional as well.

Reply:
We try to avoid having blocks w/o required values- Terraform doesn't handle them well when their subfields are unset, returning a list with a wrapped nil inside. The expander code as written right now would cause a provider crash when `advanced_machine_features {}` is specified.

We should keep the field required imo. If more attributes are added to the block in the future, we can always modify it then.

Reply:
Thanks for explaining, switched to optional.